src/box/lua/config/instance_config.lua

```diff
@@ -188,15 +188,41 @@ local function instance_uri(self, iconfig, advertise_type, opts)
 end
 
 local function apply_vars_f(data, w, vars)
-    if w.schema.type == 'string' and data ~= nil then
-        assert(type(data) == 'string')
-        return (data:gsub('{{ *(.-) *}}', function(var_name)
-            if vars[var_name] ~= nil then
-                return vars[var_name]
+    local function subst(str)
+        return (str:gsub('{{%s*(.-)%s*}}', function(var_name)
+            local v = vars[var_name]
+            if v ~= nil then
+                return tostring(v)
             end
             w.error(('Unknown variable %q'):format(var_name))
         end))
     end
+
+    local function traverse_inplace(val, seen)
```

Comment:
I would consider some solution on the `utils/schema.lua` level. Can we adjust `<schema object>:map()` to traverse composite types down similarly as we do that for `<schema object>:get()` (#10206)?